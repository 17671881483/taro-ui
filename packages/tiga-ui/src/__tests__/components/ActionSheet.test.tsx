import React from 'react'
import renderer from 'react-test-renderer'
import { cleanup, fireEvent, render } from '@testing-library/react'

describe('AtActionSheet 测试在 weapp 环境，h5 类似 -> ', () => {
  const OLD_ENV = { ...process.env }
  process.env.TARO_ENV = 'h5'

  afterAll(() => {
    process.env = { ...OLD_ENV } // Restore old environment
    jest.resetModules()
  })

  afterEach(cleanup)

  const AtActionSheet = jest.requireActual(
    '../../components/actionsheet'
  ).default
  const AtActionSheetItem = jest.requireActual(
    '../../components/actionsheet/ActionSheet/Item'
  ).default

  it('展示：当 isOpened 属性是 false', () => {
    const component = renderer.create(
      <AtActionSheet cancelText='取消' isOpened={false}>
        <AtActionSheetItem>按钮一</AtActionSheetItem>
        <AtActionSheetItem>按钮二</AtActionSheetItem>
      </AtActionSheet>
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('展示：当 isOpened 属性是 true', () => {
    const component = renderer.create(
      <AtActionSheet cancelText='取消' isOpened>
        <AtActionSheetItem>按钮一</AtActionSheetItem>
        <AtActionSheetItem>按钮二</AtActionSheetItem>
      </AtActionSheet>
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('展示：配置 title 和 content 属性', () => {
    const component = renderer.create(
      <AtActionSheet
        cancelText='取消'
        isOpened={true}
        title='此处是操作释义，建议20个字以内，使用陈述句'
        content='展示内容'
      >
        此处内容不展示
      </AtActionSheet>
    )

    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('交互：当有取消事件处理函数时，点击取消触发改事件，且调用关闭处理函数', () => {
    const cancelFn = jest.fn()
    const closeFn = jest.fn()

    const { getByText } = render(
      <AtActionSheet
        cancelText='取消'
        isOpened={true}
        onCancel={cancelFn}
        onClose={closeFn}
        title='此处是操作释义，建议20个字以内，使用陈述句'
        content='展示内容'
      />
    )

    fireEvent.click(getByText(/取消/i))

    expect(cancelFn).toBeCalledTimes(1)
    expect(closeFn).toBeCalledTimes(1)
  })

  it('交互：点击 mask 时，默认关闭动作面板', () => {
    const cancelFn = jest.fn()
    const closeFn = jest.fn()

    const { getByTestId } = render(
      <AtActionSheet
        cancelText='取消'
        isOpened={true}
        onCancel={cancelFn}
        onClose={closeFn}
        title='此处是操作释义，建议20个字以内，使用陈述句'
        content='展示内容'
      />
    )

    const overlay = getByTestId('at-action-sheet__overlay')

    expect(overlay.parentElement?.className).toContain('active')
    fireEvent.click(overlay)
    expect(overlay.parentElement?.className).not.toContain('active')
    expect(cancelFn).not.toBeCalled()
    expect(closeFn).toBeCalledTimes(1)
  })
})
