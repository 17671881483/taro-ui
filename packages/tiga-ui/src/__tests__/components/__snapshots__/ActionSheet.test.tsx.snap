// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AtActionSheet 测试在 weapp 环境，h5 类似 ->  展示：当 isOpened 属性是 false 1`] = `
<view
  className="at-action-sheet"
  onTouchMove={[Function]}
>
  <view
    className="at-action-sheet__overlay"
    data-testid="at-action-sheet__overlay"
    onClick={[Function]}
  />
  <view
    className="at-action-sheet__container"
  >
    <view
      className="at-action-sheet__body"
    >
      <view
        className="at-action-sheet__item at-action-sheet__item--border-bottom"
        onClick={[Function]}
      >
        按钮一
      </view>
      <view
        className="at-action-sheet__item at-action-sheet__item--border-bottom"
        onClick={[Function]}
      >
        按钮二
      </view>
    </view>
    <view
      className="at-action-sheet__footer"
      onClick={[Function]}
    >
      取消
    </view>
  </view>
</view>
`;

exports[`AtActionSheet 测试在 weapp 环境，h5 类似 ->  展示：当 isOpened 属性是 true 1`] = `
<view
  className="at-action-sheet at-action-sheet--active"
  onTouchMove={[Function]}
>
  <view
    className="at-action-sheet__overlay"
    data-testid="at-action-sheet__overlay"
    onClick={[Function]}
  />
  <view
    className="at-action-sheet__container"
  >
    <view
      className="at-action-sheet__body"
    >
      <view
        className="at-action-sheet__item at-action-sheet__item--border-bottom"
        onClick={[Function]}
      >
        按钮一
      </view>
      <view
        className="at-action-sheet__item at-action-sheet__item--border-bottom"
        onClick={[Function]}
      >
        按钮二
      </view>
    </view>
    <view
      className="at-action-sheet__footer"
      onClick={[Function]}
    >
      取消
    </view>
  </view>
</view>
`;

exports[`AtActionSheet 测试在 weapp 环境，h5 类似 ->  展示：配置 title 和 content 属性 1`] = `
<view
  className="at-action-sheet at-action-sheet--active"
  onTouchMove={[Function]}
>
  <view
    className="at-action-sheet__overlay"
    data-testid="at-action-sheet__overlay"
    onClick={[Function]}
  />
  <view
    className="at-action-sheet__container"
  >
    <view
      className="at-action-sheet__header"
    >
      此处是操作释义，建议20个字以内，使用陈述句
    </view>
    <view
      className="at-action-sheet__body"
    >
      此处内容不展示
    </view>
    <view
      className="at-action-sheet__footer"
      onClick={[Function]}
    >
      取消
    </view>
  </view>
</view>
`;
